std/haxe/io/Bytes.hx

```diff
@@ -348,6 +348,60 @@ class Bytes {
 
 	public function getString( pos : Int, len : Int, ?encoding : Encoding ) : String {
 		if( encoding == null ) encoding == UTF8;
+		if( encoding == UTF8Strict ) {
+			var s = "";
+			var b = b;
+			var fcc = String.fromCharCode;
+			var i = pos;
+			var max = pos+len;
+			// utf8-decode and utf16-encode
+			while( i < max ) {
+				var c:Int = fastGet(b, i++);
+				if( c < 0x80 ) {
+					//if( c == 0 ) break;
+					s += fcc(c);
+				} else if( c < 0xC2 ) { // invalid continuation or overlong sequence
+					throw "invalid UTF-8";
```

Comment:
I suggest turning these exceptions into classes/enums containing an error message and erroneous byte index.